tools/system_libs.py

```diff
@@ -652,6 +646,17 @@ def create_gl():
     check_call([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', 'gl.c'), '-o', o])
     return o
 
+  def create_dlmalloc(out_name, clflags):
+    o = in_temp(out_name)
+    check_call([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', 'dlmalloc.c'), '-o', o] + clflags)
+    return o
+
+  def create_dlmalloc_singlethreaded():
+    return create_dlmalloc('dlmalloc.bc', ['-O2'])
+
+  def create_dlmalloc_multithreaded():
```

Comment:
nice solution, we should do the other stuff that hit this issue similarly. e.g. tracing. cc @nickdesaulniers